xarray/test/__init__.py

```diff
@@ -215,6 +215,53 @@ def assertDataArrayAllClose(self, ar1, ar2, rtol=1e-05, atol=1e-08):
         self.assertVariableAllClose(ar1, ar2, rtol=rtol, atol=atol)
         self.assertCoordinatesEqual(ar1, ar2)
 
+def assert_xarray_equal(a, b):
+    import xarray as xr
+    ___tracebackhide__ = True
+    assert type(a) == type(b)
+    if isinstance(a, (xr.Variable, xr.DataArray, xr.Dataset)):
+        assert a.equals(b), '{}\n{}'.format(a, b)
+    elif isinstance(a, xr.core.coordinates.AbstractCoordinates):
+        assert sorted(a) == sorted(b)
+        [assert_xarray_equal(a[k], b[k]) for k in a]
+    else:
+        raise TypeError('{} not supported by assertion comparison'
+                        .format(type(a)))
+
+def assert_xarray_identical(a, b):
+    import xarray as xr
+    ___tracebackhide__ = True
+    assert type(a) == type(b)
+    if isinstance(a, xr.DataArray):
+        assert a.name == b.name
+        assert_xarray_identical(a._to_temp_dataset(), b._to_temp_dataset())
+    elif isinstance(a, (xr.Dataset, xr.Variable)):
+        assert a.identical(b), '{}\n{}'.format(a, b)
+    else:
+        raise TypeError('{} not supported by assertion comparison'
+                        .format(type(a)))
+
+def assert_xarray_close(a, b, rtol=1e-05, atol=1e-08):
```

Comment:
This is the name of the functions in [`numpy.testing`](https://docs.scipy.org/doc/numpy/reference/routines.testing.html)